cmd/object-handlers.go

```diff
@@ -716,23 +716,8 @@ func (api objectAPIHandlers) getObjectAttributesHandler(ctx context.Context, obj
 
 	objInfo, err := objectAPI.GetObjectInfo(ctx, bucket, object, opts)
 	if err != nil {
-
-		reqInfo := logger.GetReqInfo(ctx)
-		if reqInfo == nil {
-			writeErrorResponseHeadersOnly(w, errorCodes.ToAPIErr(ErrAccessDenied))
-			return
-		}
-
-		if globalIAMSys.IsAllowed(policy.Args{
-			AccountName:     reqInfo.Cred.AccessKey,
-			Groups:          reqInfo.Cred.Groups,
-			Action:          policy.ListBucketAction,
-			BucketName:      bucket,
-			ConditionValues: getConditionValues(r, "", reqInfo.Cred),
-			ObjectName:      object,
-			IsOwner:         reqInfo.Owner,
-			Claims:          reqInfo.Cred.Claims,
-		}) {
+		s3Error = checkRequestAuthType(ctx, r, policy.ListBucketAction, bucket, object)
```

Comment:
I am not sure I fully understand why this is needed, are you copying this from somewhere else ?

Reply:
Sorry about that.. It's needed because of how the endpoint behaves. 
```
If you have the s3:ListBucket permission on the bucket, Amazon S3 returns an HTTP status code 404 Not Found ("no such key") error.

If you don't have the s3:ListBucket permission, Amazon S3 returns an HTTP status code 403 Forbidden ("access denied") error.
```
[from amazons documentation on permissions](https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectAttributes.html)

Reply:
It's an weird AWS S3 behavior @vadmeste